akka-cluster-typed/src/test/scala/docs/akka/cluster/typed/DistributedPubSubExample.scala

```diff
@@ -0,0 +1,356 @@
+        import akka.actor.typed.scaladsl.adapter._
+        val mediator = DistributedPubSub(context.system.toClassic).mediator
+
+        var registry: Map[DataKey, DataType] = Map.empty
+
+        def register(key: DataKey, schema: Schema): RegistrationStatus =
+          registry.get(key) match {
+            case Some(_) =>
+              DataTypeExists(key)
+            case None =>
+              validate(schema) match {
+                case Success(vs) =>
+                  val created = DataType(key, vs, 0)
+                  registry += (key -> created)
+
+                  mediator ! DistributedPubSubMediator.Publish(RegistrationTopic, created)
+                  RegistrationSuccess(created)
+                case Failure(e) =>
+                  RegistrationFailure(key, e)
+              }
+          }
+
+        def validate(schema: Schema): Try[Schema] = {
+          Success(schema) // called, stubbed
+        }
+
+        Behaviors.receiveMessage {
+          case RegisterDataType(key, schema, replyTo, onBehalfOf) =>
+            val status = register(key, schema)
+            replyTo ! RegistrationAck(status, onBehalfOf)
+            Behaviors.same
+          case _ =>
+            Behaviors.unhandled
+        }
+      }
+      //#publisher
+
+    }
+  }
+
+}
+
+object Ingestion {
+  import Ontology._
+
+  def apply(dt: DataType, mediator: akka.actor.ActorRef): Behavior[DataEvent] = {
+    //#destination
+    Behaviors.setup { context =>
+      // register to the path
+      import akka.actor.typed.scaladsl.adapter._
+      mediator ! DistributedPubSubMediator.Put(context.self.toClassic)
+
+      idle(dt, mediator)
+    }
+    //#destination
+  }
+
+  private def idle(dt: DataType, mediator: akka.actor.ActorRef): Behavior[DataEvent] =
+    Behaviors.setup { context =>
+      Behaviors.receiveMessage {
+        case StartIngestion(_, _) =>
+          context.log.info("Processing events for {} storing.", dt.key)
+          mediator ! DistributedPubSubMediator.Publish(
+            IngestionTopic,
+            IngestionStarted(dt.key, context.self.path.toStringWithoutAddress))
+          active(dt, mediator)
+
+        case _ =>
+          idle(dt, mediator)
+      }
+    }
+
+  /** Would normally be typed more specifically. */
+  private def active(dt: DataType, mediator: akka.actor.ActorRef): Behavior[DataEvent] =
+    //#publisher
+    Behaviors
+      .supervise(Behaviors.receiveMessagePartial[DataEvent] {
+        case e: DataEnvelope if e.key == dt.key =>
+          // fanout to:
+          // validate, slice, dice, re-route, store raw to blob, store pre-aggregated/timeseries to Cs*, etc.
+          Behaviors.same
+
+        case StopIngestion(k) if k == dt.key =>
+          mediator ! DistributedPubSubMediator.Publish(IngestionTopic, IngestionStopped(dt.key))
+          // cleanup and graceful shutdown
+          Behaviors.stopped
+      })
+      .onFailure[DeathPactException](SupervisorStrategy.restart)
```

Comment:
the example isn't using `watch` so this is not needed?

Reply:
Ya, I had the watch in then removed it so I'll remove this, thanks.